hw/ip/flash_ctrl/data/flash_ctrl.sv.tpl

```diff
@@ -264,18 +280,19 @@ module flash_ctrl import flash_ctrl_pkg::*; (
 
   // software only has privilege to change creator seed when provision enable is set and
   // before the the seed is set as valid in otp
-  assign creator_seed_priv = lc_i.provision_en & ~otp_i.seed_valid;
+  assign creator_seed_priv = lc_provision_en[FlashLcCreatorSeedPriv] == lc_ctrl_pkg::On &
+                             ~otp_i.seed_valid;
```

Comment:
I have absorbed this check in the life cycle controller such that PROVISION_EN is deasserted, once ID_STATE goes from BLANK to PERSONALIZED (i.e., when the root key partition is locked down).

for the isolated flash page feature, does PROVISION_EN have to be independent of the ID_STATE? If so, I need to revert that.

Reply:
ah so if you do that, we may need to change this PROVISION_WR_EN and PROVISION_RD_EN.
Because I think i'm also using the provision_en to control read access to the seed pages for key manager.  So if PROVISION_EN does end up going low after we are personalized, it would mean we can no longer access the seed pages either.

Do you recall very originally what we said about this? I honestly can't remember.  I'm fine honestly with using another signal here, I just wanted to get some connective tissue in.

Reply:
ok splitting this out into two signals is ok from my side.

I think originally we had a condition like PROVISION_EN == ON + ID_STATE == BLANK in the spec for the write access, and then I absorbed all of that into PROVISION_EN for simplicity, since I did not know we needed more fine grained control. 

so in any case if we need finer control we would need an additional signal, and doing it via a life cycle signal sounds cleaner to me than distributing ID_STATE separately...

if we went down the road PROVISION_WR_EN and PROVISION_RD_EN, we would have that PROVISION_WR_EN = PROVISION_RD_EN == ON && ID_STATE == BLANK right?

Reply:
wait...i'm starting to remember this a bit...

The intent might have been `PROVISION_EN` = write_enable / read_enable, and this would become 0 after personalized like you said.

But then personalized (if that can be output) could then become the output that would enable the seed reads.  This would work too.  So effectively..

PROVISION_WR_EN i think is as you showed.
But PROVISION_RD_EN is that OR'd with when the state is personalized.

Reply:
Ok sounds good, I'll add that to the lc_ctrl!

Reply:
Hi, a couple of security related Qs:
Can someone move the chip to 'personalized' state without needing to sign a provisioning SW with the creator's private code signing key?
I mean, even if the certificate will not be valid, we still don't want that someone to be able to read the protected flash section.
Same line of Q regarding RMA: Can someone move the chip to RMA in order to read the protected flash section?